File: Python/heap_sort.py
def heapify(arr, n, i):
	largest = i # Initialize largest as root
	l = 2 * i + 1	 # left = 2*i + 1
	r = 2 * i + 2	 # right = 2*i + 2
	if l < n and arr[i] < arr[l]:
		largest = l
	if r < n and arr[largest] < arr[r]:
		largest = r
	if largest != i:
		arr[i],arr[largest] = arr[largest],arr[i] # swap
		heapify(arr, n, largest)
def heapSort(arr):
	n = len(arr)
	for i in range(n // 2 - 1, -1, -1):
		heapify(arr, n, i)
	for i in range(n-1, 0, -1):
		arr[i], arr[0] = arr[0], arr[i] # swap
		heapify(arr, i, 0)
l=int(input("enter length of an array")
arr = []
for x in range(l):
      v=int(input("enter the number in list"))
      arr.append(v)
heapSort(arr)
n = len(arr)
print ("Sorted array is")
for i in range(n):
	print ("%d" %arr[i])

File: C++/Heap sort.cpp
#include <iostream>
using namespace std;
 

void heapify(int arr[], int size, int i)
{
    int largest = i; 
    int l = 2 * i + 1;
    int r = 2 * i + 2; 
    if (l < size && arr[l] > arr[largest])
        largest = l;
    if (r < size  && arr[r] > arr[largest])
        largest = r;
    if (largest != i) {
        swap(arr[i], arr[largest]);
        heapify(arr, size, largest);
    }
}
void heapSort(int arr[], int size)
{
    
    for (int i = size / 2 - 1; i >= 0; i--)
        heapify(arr, size, i);
 
   
    for (int i = size - 1; i >= 0; i--) {
        
        swap(arr[0], arr[i]);
 
       
        heapify(arr, i, 0);
    }
}
 

void printArray(int arr[], int size)
{
    for (int i = 0; i < size; ++i)
        cout << arr[i] << " ";
    cout << "\n";
}
int main()
{
    int size;
	printf("Enter the Size:");
	scanf("%d",&size);
	int arr[size];
	printf("Enter the Array:");
	int i;
	for(i=0;i<size;i++)
	{
		scanf("%d",&arr[i]);
	}
    heapSort(arr, size);
 
    cout << "Sorted array is \n";
    printArray(arr, size);
}
 


   
